cmd/abapEnvironmentCheckoutBranch.go

```diff
@@ -150,3 +146,85 @@ func triggerCheckout(repositoryName string, branchName string, checkoutConnectio
+		log.Entry().Info("It seems like you have specified a dedicated repository configuration file but also an in-line configuration for the repository or branch to be pulled.")
+		log.Entry().Info("Please note that in this case the dedicated repositories configuration file will be handled only.")
+	}
+	return nil
+}
+
+func checkoutBranchesFromConfigFile(repositoriesFile string, checkoutConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (err error) {
+	if fileExists(repositoriesFile) {
+		fileContent, err := ioutil.ReadFile(repositoriesFile)
+		if err != nil {
+			return fmt.Errorf("Failed to read repository configuration file: %w", err)
+		}
+		var repositoriesFileConfig []abaputils.Repository
+		var result []byte
+		result, err = yaml.YAMLToJSON(fileContent)
+		if err == nil {
+			err = json.Unmarshal(result, &repositoriesFileConfig)
+		}
+		if err != nil {
+			return fmt.Errorf("Failed to parse repository configuration file: %w", err)
+		}
+		if len(repositoriesFileConfig) == 0 {
+			return fmt.Errorf("Failed to parse repository configuration file: %w", errors.New("Empty or wrong configuration file. Please make sure that you have correctly specified the branches in the repositories to be checked out"))
+		}
+		for _, repositoryFileConfig := range repositoriesFileConfig {
+			err = handleCheckout(repositoryFileConfig, checkoutConnectionDetails, client, pollIntervall)
+			if err != nil {
+				return fmt.Errorf("Failed to checkout branch: %w", err)
+			}
+		}
+	} else {
+		return fmt.Errorf("Failed to read repository configuration file: %w", errors.New(repositoriesFile+" is not a file or doesn't exist"))
+	}
+	return err
+}
+
+func handleCheckout(repo abaputils.Repository, checkoutConnectionDetails abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (err error) {
+	if reflect.DeepEqual(abaputils.Repository{}, repo) {
+		return fmt.Errorf("Failed to read repository configuration: %w", errors.New("Eror in configuration, most likely you have entered empty or wrong configuration values. Please make sure that you have correctly specified the branches in the repositories to be checked out"))
+	}
+	startCheckoutLogs(repo.Branch, repo.Name)
+
+	uriConnectionDetails, err := triggerCheckout(repo.Name, repo.Branch, checkoutConnectionDetails, client)
+	if err != nil {
+		return fmt.Errorf("Failed to trigger Checkout: %w", errors.New("Checkout of "+repo.Branch+" for software component "+repo.Name+" failed on the ABAP System"))
+	}
+
+	// Polling the status of the repository import on the ABAP Environment system
+	status, err := abaputils.PollEntity(repo.Name, uriConnectionDetails, client, pollIntervall)
+	if err != nil {
+		return fmt.Errorf("Failed to poll Checkout: %w", errors.New("Status of checkout action on repository"+repo.Name+" failed on the ABAP System"))
+	}
+	if status == "E" {
```

Comment:
I vaguely remember a PR from @CCFenner where these status codes were moved to constants.

Reply:
Introduced the constant `statusCheckoutFail` for that